### PR TITLE
[Java] concept basics exercise

### PR DESCRIPTION
**Pull request by mirkoperillo**
I've done the porting of `basics` exercise from C# track.

Two Java track peculiarity:
* I introduced the same gradle build process we have in v2,
* I used assertj as assertion library as decided some time ago.

### CODE REVIEW

**Comment by neenjaw** [comment]
Because your branch exists on your own fork, the the CI formatter using the `/format` command doesn't work (I think). (Is that correct @SaschaMann @ErikSchierboom?)

**Comment by ErikSchierboom** [comment]
> It should work, and at least during testing it did, as long as "allow edits by maintainers" is checked.

Yep, this is correct. @mirkoperillo an alternative is to run `./bin/format.sh` (or `./bin/format.cmd` on Windows) from the root directory of this repo.

File: languages/java/exercises/concept/basics/.docs/introduction.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+// count = false;
+```
+
+Java is an [object-oriented language][object-oriented-programming] and requires all functions to be defined in a _class_. The `class` keyword is used to define a class.
+
+```java
+class Calculator
+{
+    // ...
+}
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/basics/.docs/introduction.md:23]: Should this be formatted as follows:

```suggestion
class Calculator {
    // ...
}
```

File: languages/java/exercises/concept/basics/.docs/introduction.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+A function within a class is referred to as a _method_. Each method can have zero or more parameters. All parameters must be explicitly typed, there is no type inference for parameters. Similarly, the return type must also be made explicit. Values are returned from functions using the `return` keyword. To allow a method to be called by other classes, the `public` access modifier must be added.
+
+```java
+class Calculator
+{
+    public int add(int x, int y)
+    {
+        return x + y;
+    }
+}
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/basics/.docs/introduction.md:35]: Should this be formatted as follows:

```suggestion
class Calculator {
    public int add(int x, int y) {
        return x + y;
    }
}
```

File: languages/java/exercises/concept/basics/.docs/after.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+Java is an object-oriented language and requires all functions to be defined in a _class_, which are defined using the [`class` keyword][classes]. A function within a class is referred to as a _method_. Each [method][methods] can have zero or more parameters. All parameters must be explicitly typed, there is no type inference for parameters. Similarly, the return type must also be made explicit. Values are returned from functions using the [`return` keyword][return]. To allow a method to be called by other classes, the `public` access modifier must be added.
+
+```java
+class Calculator
+{
+    public int add(int x, int y)
+    {
+        return x + y;
+    }
+}
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/basics/.docs/after.md:26]: Should this be formatted as follows:

```suggestion
class Calculator {
    public int add(int x, int y) {
        return x + y;
    }
}
```

File: languages/java/exercises/concept/basics/.docs/after.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+If a method does not use any class _state_ (which is the case in this exercise), the method can be made _static_ using the `static` modifier. Similarly, if a class only has static methods, it too can be made static using the `static` modifier.
+
+```java
+static class Calculator
+{
+    public static int multiply(int x, int y)
+    {
+        return x * y;
+    }
+}
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/basics/.docs/after.md:44]:
```suggestion
static class Calculator {
    public static int multiply(int x, int y) {
        return x * y;
    }
}
```

**Comment by ErikSchierboom** [comment]
@mirkoperillo It worked out in the end?

**Comment by mirkoperillo** [comment]
@ErikSchierboom I confirm I'm not able to use format script
if I run from root folder of project `./bin/format.sh` I have always this output
```
npx: installed 1 in 1.353s
Unexpected token *
```
npx version (the last one I think): `10.2.2`

**Comment by mirkoperillo** [comment]
@ErikSchierboom I resolved my problem with `format.sh` script.
The version of nodejs on my pc was very old.
I upgraded to the last LTS of node and the script works fine.

**Comment by ErikSchierboom** [comment]
Great! Thanks for letting me know.

**Pull request merged**